x-pack/legacy/plugins/apm/server/lib/services/get_service_node_metadata.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { idx } from '@kbn/elastic-idx';
+import { Setup } from '../helpers/setup_request';
+import {
+  HOST_NAME,
+  SERVICE_NAME,
+  SERVICE_NODE_NAME,
+  CONTAINER_ID
+} from '../../../common/elasticsearch_fieldnames';
+import { NOT_AVAILABLE_LABEL } from '../../../common/i18n';
+
+export async function getServiceNodeMetadata({
+  serviceName,
+  serviceNodeName,
+  setup
+}: {
+  serviceName: string;
+  serviceNodeName: string;
+  setup: Setup;
+}) {
+  const { client, config } = setup;
+
+  const query = {
+    index: config.get<string>('apm_oss.metricsIndices'),
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            { term: { [SERVICE_NAME]: serviceName } },
+            { term: { [SERVICE_NODE_NAME]: serviceNodeName } }
```

Comment:
You should also add a filter for metric documents:
```js
{ term: { [PROCESSOR_EVENT]: 'metric' } },
```

By `apm_oss.metricsIndices` will be `apm-*` by default so you'll get all documents back that matches `serviceName` and `serviceNodeName`.

I was thinking if we should be smarter about this. Eg adding a new property like:

```
const query = {
 apmIndex: 'metric',
 body: {...}
}
```

Which then applies the correct indices and filter.

I went over all our queries and only found a single query missing the `processor.event` filter:
https://github.com/elastic/kibana/blob/027476f9a7f9901e63ebd2c15fa13004a5a69510/x-pack/legacy/plugins/apm/server/lib/transactions/breakdown/index.ts#L82-L87

Mind fixing that while you are at it? 🙏

Reply:
Thanks, I didn't know it defaulted to `apm-*`, good to know. Added `processor.event` filters in both places.